### ISSUE TITLE
Dev Hangouts

### ISSUE THREAD

**Issue by robshep** [NorthWalesTech/NorthWalesTech#55]
During the event of Sept 29th, I (RGS) proposed a new event style: "_unstructured developer hangouts_" which had great response for potential attendees.

Having thought about it for a week or so, Here is my brain dump for this event style.

**NWT objectives**
* Primarily software development oriented.
* to create opportunities to find new collaborators & friends
* to offer a friendly space for inviting newbie developers (students, home hackers, (accompanied) kids) who can just shoulder surf the more experienced amongst us.
* to provide a friendly space for demonstrating projects/ideas in the company of likeminded people, and to get feedback on prototypes etc.

**Example Attendee Aims**
* to hack on work/hobbies in the company of other devs
* to socialise
* to ask for help from other attendees
* to offer help and support to other attendees
* to learn
* to read
* to discuss NWT & other events(TODO catch recursion; TODO meta alert )

**Proposed Naming**
* /dev/random  
* or NWT:/dev/random
* suggest a separate event numbering scheme, E.g. date based to make it easier to track them, E.g. %Y%M-%N or %Y-%j

**Event Structure**
* Initially once a fortnight
* first in month on a weekday evening
* second in month on a weekend morning/afternoon
* monitor attendance and get feedback.
* some sort of whiteboard space, possibly a facebook group post, where people can state their aims before attending or when they arrive, whatever they may be - _really encourage this one._  
* Allow people to come and go as they like but try and avoid cliques/silos and encourage openness and inclusion (_reminder: friendly space for inviting newbies_)

**Event admin**
* Minimal NWT organiser effort, for 50-100 times a year.
* Switch up the venues as usual
* Switch up the evening days, but keep a regular time, (_leave space for dinner or get pizza?_)
* Switch up the weekend days, but maybe experiment with mornings/afternoons/evenings (4hr slot?)
* at minimum have some bottled water.  Maybe ST can sponsor us a dozen cases of bottled water and we can deposit them around various people/offices. 
* Maybe try some interesting venues: first hydro VC, bethesda library, (_Good Internet is the principle requirement_)
* Evenings at parc menai will be ok, but being in Bangor somewhere for the weekend ones will be better for finding more food/refreshments

**Fun stuff**
Maybe have some sort of game, E.g. attendees can propose people who have helped them in some way, 10 nominations gets a t-shirt or some other swag.  
(_Need to think of more reasons to approach WG for swag_)

Comments please!

**Comment by carwyn** [comment]
Basically this https://en.m.wikipedia.org/wiki/Unconference

Probably a BarCamp in the first instance but try a number of the proven formats. A variant on a Code Jam is another one, thinking of hybridising BarCamp and CodeJam in November as a pilot. Power sockets and WiFi are the main limitations.

**Comment by carwyn** [comment]
The Code Jam elements are more like a music jam session than a hackathon. Themed tables and rotation of people (see speed feeling) forces people to meet and talk to new faces. FLOSS UK and others have been doing these kinds of events for years, largely cloned from the US.

**Comment by robshep** [comment]
I was thinking even less than these.....

The unconference and to some extent the barcamp still seem to be focussed around having "sessions" of group discussions/speaking&listening.
I gather people are more interested in just having a space for "hanging out & 'doing' together" rather than the talking and listening style of activities. 

I think we can discover the speakers & topics for lightning (or larger) talks at even more unstructured events where we just let people get on with whatever they might want to do, but with a bit more transparency about what people are up to.  

The only codejam I can find is the Google competitive coding event.  Not something I've really considered as it's a bit too elitist for what I think NWT should be about on the whole. 

The themed tables sounds OK as a one-off, but wouldn't sound appealing to me if I'm in the mood for just coding something up, but fine as an experiment.

I think just enough "structure" so they can actually happen, and people get something out of it, and to make them as inclusive as possible.

**Comment by davehun** [comment]
> On 7 Oct 2016, at 11:03, robshep <notifications@github.com> wrote:
> 
> I was thinking even less than these.....
> 
> The unconference and to some extent the barcamp still seem to be focussed around having "sessions" of group discussions/speaking&listening.
> I gather people are more interested in just having a space for "hanging out & 'doing' together" rather than the talking and listening style of activities.
> 

Isn’t this what fab lab is supposed to do ?
hack my heating .. have lots of code / things to build

**Comment by robshep** [comment]
Yes, fablab is certainly for making/building physical things, but I'm proposing 'doing' as principally software dev oriented.

(Unless i've misunderstood what the fablab is intended for)

Somebody mentioned last time that the hardware/electronics/making side of things can be quite intimidating if you've no experience in that area so I'm keen to explore something that is focussed on people's software development projects/aims.

**Comment by jonbullock** [comment]
👍 

Has a lot of similarities to a Hackergarten in my opinion but a Hackergarten does have a predefined list of projects to work on usually and this sounds like it shouldn't be that formal...

**Comment by davehun** [comment]
> On 7 Oct 2016, at 11:28, robshep <notifications@github.com> wrote:
> 
> Yes, fablab is certainly for making/building physical things, but I'm proposing 'doing' as principally software dev oriented.
> 
> (Unless i've misunderstood what the fablab is intended for)
> 
> Somebody mentioned last time that the hardware/electronics/making side of things can be quite intimidating if you've no experience in that area so I'm keen to explore something that is focussed on people's software development projects/aims.
> 
> —
> 

so soft lab. A software side of things. There is the space there and having both hardware and software hacking going on at the same time would be nice ..

**Comment by tamslinn** [comment]
I think should be unstructured, turn up with your laptop and work on whatever you want, and be able to chat to other people about stuff if you want to/get help if you're a learner. Too much structure might put people off, although designated nwt people could be on hand to make sure new people don't get left out etc, facilitate introductions.

**Comment by tomusher** [comment]
I like this idea - sounds like a session-based co-working space. Would be great to help limit the social isolation issues that come from remote work.

**Comment by carwyn** [comment]
Code Jam, Hackathon, Hackergarten and Code Sprints are all similar and not elitist in general and more about fun and good causes. The NHS ones area good examples or the Space Apps one we said we'd join next year. I'm actually spending today talking to a company a possible transport and health related hackathon later this year.

Google poached the Code Jam moniker a bit, Pi Jams and Music Session Jams are more like what I mean, these are very much along what Rob describes. There is an issue with completely open ended sessions though as that makes the assumption that everyone has something to work on and also tends to encourage cliques to form and not always good for beginners.

The fablab is a good model where by there are set projects as well as freedom to work on anything you want. What we are describing here is a software lab in some respects.

We're not a co-working space, there are issues with using free venues for work on commercial projects too.

**Comment by carwyn** [comment]
Fablabs and softlabs running alongside each other would be the ideal for me. Encouraging creation on both fronts. We have room to try many different event types. We've been asked for:

* Slide roulette.
* Barcamps.
* Structured workshops.
* Freeform sessions (hardware and software).
* Hackathons.
* Series of teaching sessions on a specific topic.
* PiJams.
* Beginner events.
* Famous speakers.
* Business networking events.
* Field trips.
* Evening and weekend access to the fablab.

**Comment by TheInvader360** [comment]
I really like the idea of getting together with a group of local people, with a shared area of interest, at a suitable location, with laptops in hand, and ending up possibly doing something awesome.

I'd really appreciate the opportunity to hang out with more experienced and accomplished developers working on the stuff they love, stuff they're passionate enough about to be doing it just for fun. I'm sure it would be very different from the average "day job" experience.

When we were in the CoLab last week, I couldn't help but think it would be an excellent hackerspace. I figure the FabLab is more about fabrication, hardware projects, electronics, embedded systems, etc. Somewhere like the CoLab, just a nice sized room with desks and whiteboards and internet access might be more suited to software projects? I understand that you can't just take over the CoLab as a permanent residence, but if you could secure access to a number of suitable locations locally that we could use in rotation, that would be great! The minimum requirements are pretty low really, internet access, ideally some desks, and with a bit of luck some people turning up :)

**Comment by carwyn** [comment]
The Colab and Media labs are for exactly this purpose, they are also part of the "fablab" membership. Our last event was the very first third party evening event in the Colab and the next one is in the same place to try and enable regular access to this space.

What I think would work is to have one "home base" event in Pontio, Ty Menai or maybe CBC every month and the second one in a month in a remote location. The following locations have been requested:

* Studio Made Denbigh
* Artisans Collective Prestatyn
* Conwy Business Centre
* Studio 16 Llandudno
* Caban Brynsiencyn.
* Lodge Dynorwig.
* Neuadd Ogwen
* Bethesda Library
* Caernarfon
* Blaenau Festiniog
* CAT
* Llangollen
* Holyhead

**Comment by carwyn** [comment]
For co-working see:

http://indycube.cymru/

http://www.16trinity.co.uk/drum-coworking-hotdesking-16-trinity/

Indycube have met up with us already and are happy to collaborate.

**Comment by tamslinn** [comment]
Not suggesting co-working, more working on personal projects, learning new software etc, in a social atmosphere :)

**Comment by robshep** [comment]
Yeah, certainley not a work environment - it'll be outside of normal work hours in any case, and whatever shape it takes, it won't be quiet enough to do any actual work :)

**Comment by carwyn** [comment]
See also:

* https://en.wikipedia.org/wiki/Homebrew_Computer_Club
* https://en.wikipedia.org/wiki/SuperHappyDevHouse
* https://en.m.wikipedia.org/wiki/Hacker_Dojo

**Comment by robshep** [comment]
Good find Carwyn 😺  

The SuperHappyDevClub gets my vote.

Or maybe Clwb datblygwr uwch-hapus or something

**Comment by carwyn** [comment]
How would you all feel about a *recommendation* (not enforced by us) that activities be done under one of:

http://choosealicense.com/

See also: http://fabfoundation.org/the-fab-charter/

One of the reasons for this is to reduce issues with IP ownership from material created through informal collaboration and potential issues with state funding claims. By making the recommendation we can raise awareness of the issue and absolve ourselves at the same time. This has happened in other similar groups.

**Comment by carwyn** [comment]
I'd like to reclaim the **Code Jam** name from Google while we are at it as they've obviously misappropriated it:

https://en.wikipedia.org/wiki/Jam_session

**Comment by carwyn** [comment]
There was a SuperHappyDevClub in Cambridge http://shdc.pbworks.com/w/page/4439183/FrontPage until a few years back.

I like /dev/random an awful lot but it's impossible to search for :( I agree with different naming much like the field trips. I'd suggest dropping the date and just using the *event type name* repeated using the Meetup.com repeat mechanism.

Name Suggestions:

* *Jam Based* Jam sessions are well understood from Raspberry Jams and Music Jam Sessions. A mix of these to direct the type of activity could be an option.
  * **Code Jam**: conflicts with Google event but that may work in our favour in time. May not be just coding activities though.
  * **Dev Jam**: Similar to Code Jam, may be too specific.
  * **Tech Jam**: fits with NWT name, broad enough to cover everything. 
* **Soft(ware) Lab**: Meshes nicely with FabLab, CoLab, MediaLab. Possibly too Pontio related.
* **SuperHappyDevClub**: could take it over :)

Computer Science at Bangor have the *Software Hut* course https://www.bangor.ac.uk/cs/undergraduate-modules/ICP-2302 .. which is also a thing at other Universities.

**Comment by carwyn** [comment]
Created poll on location:

https://www.facebook.com/groups/NorthWalesTech/permalink/1495195263830526/

.. will probably get the same kind of result as last time. Ty Menai isn't that popular as people don't want to socialize where they work on a regular basis. Less valid for places we've not been as our member density in those areas is lower.

**Comment by tamslinn** [comment]
"Hack" has security connotations for me. Jam and club based names get my vote.  Code jam is a ideal unless we'd get in trouble with Google...

**Comment by carwyn** [comment]
Planets are aligning, have a plan forming for the early November event. I think a hybrid initially (as mentioned in email a few weeks back to some of you) of an Unconference and Code/Tech Jam would work best with a view to making most of our events of this type to reduce organisational overhead, effectively make all but the bigger events JIT based along these lines.

There are some characteristics I think that are critical:

* **Inclusive**: Not everyone is a coder, we want to attract budding techies too. Graphic designers, digital musicians etc. This openness is one of the reasons for our success, it's also the most likely way to encourage innovative collaboration. I think we can cater for this within the loose structure.
* **Less Focus On Physical**: More software and things that don't need more than a laptop. Easier to organise. Not exclusive though, Micro:bits for example would work very well.
* **Encourage Communication**: Allowing for a small number of show and tell type elements will encourage topics of conversation and give opportunities for people to practice speaking in front of smaller groups. [Speed Geeking](https://en.wikipedia.org/wiki/Speed_geeking) is a twist on this.
* **Always Have Something To Do**: Have a default activity for those without their own plans or projects.
* **Beginner Friendly**: Always have provision for beginners.
* **Learn From Others**: There is a wealth of information from the tech community on how to structure events, we should try many of them. The teaching community is also a huge resource here. Cramming people in a room and expecting them to interact doesn't always work :) There are certainly techniques that can bring a lot more out of people.

**Comment by carwyn** [comment]
Tada: http://www.meetup.com/NorthWalesTech/events/235312802/

**Comment by carwyn** [comment]
Hopefully we've managed to keep everyone happy with the event design :) The first one is a little more structured to introduce the concepts and some of the possibilities.

The intro will cover a bit about the origin of the SuperHappyDevClub subtitle which links back to the Hackers in Space event and the spirit we're keen to kindle with all this. I'll then introduce Unconference/Self Organising events and do a whiteboard schedule and identify breakout groups. The Kata will be accessible to everyone regardless of ability or language (took some thinking!). Even those in the breakout groups will be able to dip in and out.

Facilities wise we are quite spoilt:

* 2x 62" TV screens on wheels (used at Android OS event)
* 1x smaller screen at the head of one of the table clusters.
* 1x projector (used at rocket event)
* Lots of bean bags.
* Mobile whiteboards.
* Power sockets everywhere!
* I'll sort out a batch of wifi guest accounts for those without Uni or FabLab accounts.
* Sponsored PIZZA!

At the moment I'm liking the one "home" event and one "away" event a month, adjusting style depending on content and interest but always having one Jam Session at least. Hopefully we can get to a weekly slot. I have the ok from API to do this. Really keen to go after the HE and FE venues for this for pure logistical reasons (or any others with these easy access characteristics).

**Comment by tamslinn** [comment]
It sounds great! I'm happy :)

**Comment by carwyn** [comment]
Need to capture this in a Wiki Page then close it off I think, we have a good base here.
